scipy/optimize/tests/test_linprog.py

```diff
@@ -1403,8 +1411,18 @@ class TestLinprogIPDense(LinprogIPTests):
     options = {"sparse": False}
 
 
+if has_cholmod:
+    class TestLinprogIPSparseCholmod(LinprogIPTests):
+        options = {"sparse": True, "cholesky": True}
```

Comment:
This is all a bit indirect. http://scipy.github.io/devdocs/optimize.linprog-interior-point.html#optimize-linprog-interior-point suggests that there's also `scikit-sparse` usage for dense problems, which doesn't seem to cover this. Wouldn't it make sense to allow forcing a particular solver?

Reply:
> ...suggests that there's also scikit-sparse usage for dense problems, which doesn't seem to cover this
Oops. No, that is a mistake in the documentation. For dense problems, scikit-sparse is irrelevant, and I haven't made any changes to the way dense problems are handled. I'll fix that.
> Wouldn't it make sense to allow forcing a particular solver?
You can force a particular solver. 
"options can be set to bypass solvers that are failing. Setting cholesky=False skips to solver 2, sym_pos=False skips to solver 3, and lstsq=True skips to solver 4 for both sparse and dense problems."

I'll fix that documentation.

Reply:
> This is all a bit indirect.

Perhaps you are referring to the use of three separate boolean options `cholesky`, `sym_pos`, and `lsqr` to change the solver?

I would agree that is horrible now - but it's the way it has been. I had intend to change that to a single option: `solver` with permissible values 1 - 4 after 1.3, but I suppose I could try to hack it ASAP if you'd prefer.

Reply:
> Perhaps you are referring to the use of three separate boolean options cholesky, sym_pos, and lsqr to change the solver?

indeed

> I had intended to change that to a single option: solver with permissible values 1 - 4 after 1.3,

`solver='umfpack'` etc., but yeah that's what I meant. 

> I suppose I could try to do that ASAP if you'd prefer.

it's not that urgent, maybe just open an issue for it.

Reply:
Added to #9671. 
Shall we merge this PR?

Reply:
@rgommers @tylerjereddy should this go into 1.3 or wait for the `solver` option?

Reply:
just testing ....

Reply:
so much fun installing suitesparse locally and making those things build ....

Reply:
I didn't have the pleasure of building myself. `conda` to the rescue!